exchanger/src/main/kotlin/com/d3/exchange/exchanger/RateStrategies.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright D3 Ledger, Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.d3.exchange.exchanger
+
+import com.google.gson.JsonParser
+import java.math.BigDecimal
+
+/**
+ * Rate retrieval interface
+ */
+interface RateStrategy {
+
+    /**
+     * Method for retrieving relevant exchange rate for 'non standard' assets
+     */
+    fun getRate(from: String, to: String): BigDecimal
+}
+
+/**
+ * Rate strategy based on http querying of data collector service
+ */
+class DcRateStrategy(
+    private val baseRateUrl: String,
+    private val baseAssetId: String,
+    private val rateAttribute: String
+) : RateStrategy {
+
+    private val parser = JsonParser()
+
+    override fun getRate(from: String, to: String): BigDecimal {
+        val fromRate: BigDecimal =
+            if (from == baseAssetId) {
+                BigDecimal.ONE
+            } else {
+                getRateFor(from)
+            }
+        val toRate: BigDecimal =
+            if (to == baseAssetId) {
+                BigDecimal.ONE
+            } else {
+                getRateFor(to)
+            }
+        return fromRate.divide(toRate)
```

Comment:
Can `toRate` be zero?

Reply:
No, but if it happens an exception leading to a rollback will be thrown